sympy/assumptions/handlers/matrices.py

```diff
@@ -38,6 +38,8 @@ def MatMul(expr, assumptions):
         factor, mmul = expr.as_coeff_mmul()
         if all(ask(Q.symmetric(arg), assumptions) for arg in mmul.args):
             return True
+        if ask(Q.diagonal(expr), assumptions):
```

Comment:
@jksuom may be it is related to "the new-style assumptions don't support this, only the old ones do that kind of thing."
https://groups.google.com/forum/?fromgroups#!searchin/sympy/inference%7Csort:relevance/sympy/sOeIU6nz-gM/DoVGj1Ka_WMJ